src/Aspire.Hosting/Publishing/DockerContainerRuntime.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.Publishing;
+
+internal sealed class DockerContainerRuntime(ILogger<DockerContainerRuntime> logger) : IContainerRuntime
+{
+    private async Task<(int ExitCode, string? Output)> RunDockerBuildAsync(string contextPath, string dockerfilePath, string imageName, CancellationToken cancellationToken)
+    {
+        var startInfo = new ProcessStartInfo
+        {
+            FileName = "docker",
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+        };
+
+        startInfo.ArgumentList.Add("build");
+        startInfo.ArgumentList.Add("--file");
+        startInfo.ArgumentList.Add(dockerfilePath);
+        startInfo.ArgumentList.Add("--tag");
+        startInfo.ArgumentList.Add(imageName);
+        startInfo.ArgumentList.Add("--quiet");
+        startInfo.ArgumentList.Add(contextPath);
+
+        logger.LogInformation("Running Docker CLI with arguments: {ArgumentList}", startInfo.ArgumentList);
+
+        using var process = Process.Start(startInfo);
+
+        if (process is null)
+        {
+            throw new DistributedApplicationException("Failed to start Docker CLI.");
+        }
+
+        await process.WaitForExitAsync(cancellationToken).ConfigureAwait(false);
+
+        if (process.ExitCode != 0)
+        {
+            var stderr = process.StandardError.ReadToEnd();
+            var stdout = process.StandardOutput.ReadToEnd();
+
+            logger.LogError(
+                "Docker CLI failed with exit code {ExitCode}. Output: {Stdout}, Error: {Stderr}",
+                process.ExitCode,
+                stdout,
+                stderr);
```

Comment:
Does this show up nicely in the CLI?

Reply:
No not yet, I haven't figured out how I want to stream back errors yet.